design/ceph/object/store.md

```diff
@@ -712,3 +712,64 @@ For external CephCephObjectStores (i.e., when `spec.gateway.externalRgwEndpoints
 vhost-style addressing should be configured on the host cluster, and `hosting.dnsNames` is
 irrelevant. The default `advertiseEndpoint` for external CephObjectStores is the first entry in the
 `spec.gateway.externalRgwEndpoints` list, which users should be able to override if desired.
+
+### Advanced user configuration
+
+RGW is a very active project with many existing features and regular new feature additions. Rook
+maintainers have a hard time catching up with RGW's latest features. In an effort to help users who
+want access to rare and new features, Rook can allow users to set arbitrary Ceph configs in their
+CephObjectStore CR. RGW config reference: https://docs.ceph.com/en/reef/radosgw/config-ref/
+
+This is an advanced feature. It allows users to modify many untested aspects of the RGW and object
+store. Any user configs that break the object store should be a responsibility of the user.
+
+```yaml
+spec:
+  metadataPool: # ...
+  dataPool: # ...
+  gateway: # ...
+
+  cephConfig:
```

Comment:
Would this map be merged with `cephcluster.spec.cephConfig`? IOW - would keys set on the cephobjecstore override global values in the global cephConfig (if someone had set `client.rgw.<cephobjectstore name>.a`) or would the operator treat them as completely separate sets of config and try to apply them independently? Either is probably a defensible semantic... trying to think through how this would work in practice.

Reply:
This is an important question, thanks! I'll update my doc to address this. 

My intent is that this won't really be **aware** of the `cephcluster.spec.cephConfig` at all. That could mean some configs that are set in both places might contend with each other. In the future, we could do some work to address that, but for immediate MVP I think the complexity is a bit too much work for something that would be easiest to mitigate via documentation.

Reply:
That seems completely reasonable to me.  This is a "you should know what your doing" type of feature.